tests/cpp/test_multidevice_tutorial.cpp

```diff
@@ -0,0 +1,367 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <multidevice/communicator.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+
+namespace nvfuser {
+
+class MultiDeviceTutorial : public MultiDeviceTest {
+ protected:
+  static void SetUpTestSuite() {
+    verbose_ = getNvFuserEnv("TUTORIAL_VERBOSE");
+  }
+
+ protected:
+  static bool verbose_;
+};
+
+bool MultiDeviceTutorial::verbose_ = false;
+
+// The first object we need to introduce is the class `Communicator` which is
+// a convenience that is nvFuser's interface to runtime distributed setting
+// and backend. Communicator's setup is expansive (because all IP addresses
+// need to be exchanged to all ranks) and occupies a port. Therefore, we
+// instantiate one Communicator globally and use it accross all tests. There
+// is no benefit in having multiple communicator, and this class is thought to
+// have a singleton instance. This communicator can be accessed through
+// MultiDeviceTest::communicator_
+TEST_F(MultiDeviceTutorial, CommunicatorAndC10d) {
+  Communicator* communicator = communicator_;
+
+  // To check if distributed setting is available:
+  if (communicator->is_available() == false) {
+    GTEST_SKIP() << "distributed setting not available";
```

Comment:
Could this be potentially a common error message users might encounter? Could we give users some more actionable error messages?

Reply:
I moved this check to the class's setup and provided a more actionavle error message.